src/pages/payment-methods/index.js

```diff
@@ -82,14 +84,79 @@ const InnerDiv = styled.div`
     overflow-x: auto;
     overflow-y: visible;
 `
+
+const getClientResidence = () => {
+    const domain = getDomain()
+    const client_information = getClientInformation(domain)
+    return client_information ? client_information.residence : ''
+}
+
+const getClientCountry = () => {
+    let current_client_country = ''
+    if (isLoggedIn()) {
+        current_client_country = getClientResidence()
+    } else {
+        const [website_status] = useWebsiteStatus()
+        current_client_country = website_status?.clients_country || ''
+    }
+    return current_client_country.toLowerCase()
+}
+const getPaymentsBasedOnCountry = (current_client_country) => {
+    const payments_data_array = payment_data.map((category) => {
+        const filteredData = category.data.filter((item) => {
+            if (item.countries.included.length) {
+                const includedCountries = item.countries.included.map((i) => i.toLowerCase())
+                if (includedCountries.includes(current_client_country)) {
+                    return true
+                } else if (includedCountries.includes('eu')) {
+                    return isEuCountry(current_client_country)
+                }
+                return false
+            } else if (item.countries.excluded.length) {
+                const excludedCountries = item.countries.excluded.map((i) => i.toLowerCase())
+                if (excludedCountries.includes(current_client_country)) {
+                    return false
+                }
+                if (excludedCountries.includes('eu')) {
+                    return !isEuCountry(current_client_country)
+                }
+                return true
+            }
+            return true
+        })
+        return {
+            ...category,
+            data: filteredData,
+        }
+    })
+    return payments_data_array.filter((item) => item.data.length)
+}
+
 const DisplayAccordion = (locale) => {
-    const { is_eu_country, crypto_config } = React.useContext(DerivStore)
-    return (
-        <Accordion has_single_state>
-            {payment_data.map((pd, idx) => {
-                if (pd.is_crypto && is_eu_country) {
-                    return []
-                } else
+    const { crypto_config } = React.useContext(DerivStore)
+    const current_client_country = getClientCountry()
+    const [show_table, setShowTable] = useState(false)
+
+    const paymentsBasedOnCountry = getPaymentsBasedOnCountry(current_client_country)
+
+    if (!paymentsBasedOnCountry.length) {
+        return (
+            <>
```

Comment:
You don't need fragments if it's just one parent component.

Reply:
done